packages/medusa/src/api/routes/store/collections/index.ts

```diff
@@ -22,11 +22,12 @@ export default (app) => {
   return app
 }
 
-export const defaultStoreCollectionRelations = ["products"]
+export const defaultStoreCollectionRelations = []
```

Comment:
defaultStoreCollectionRelations is not used and is not passed to services in controllers. To avoid confusion, `product` has been moves to allowedFields. This change should have no effect.